### PR TITLE
relative color logic, button implementation

### PR DESCRIPTION
**Pull request by adids1221**
## Description
Relative color logic based on background of the component.
Implementation in Button component.

## Changelog
Relative color logic based on background of the component.
Implementation in Button component.

## Additional info
MADS-4291 / 4455

### CODE REVIEW

**Comment by M-i-k-e-l** [comment]
@adids1221 Assigning to you as you've said it's not ready

File: src/style/colors.ts [hunk 1]
```diff
@@ -341,6 +341,11 @@ export class Colors {
   isDesignToken(color?: DesignToken) {
     return !!(color?.semantic || color?.resource_paths);
   }
+  getRelativeColor(backgroundColor: string) {
```
Comment (Inbal-Tish) [inline src/style/colors.ts:344]: The relative color is the content color, the name should reflect that

File: src/components/button/index.tsx [hunk 2]
```diff
@@ -103,21 +103,32 @@ class Button extends PureComponent<Props, ButtonState> {
...
+      backgroundColor,
+      modifiers
+    } = this.props;
     const {color: modifiersColor} = modifiers;
     const isLink = this.isLink;
+    const contentColor = Colors.getRelativeColor(this.getBackgroundColor());
+
+    let color: string | undefined = contentColor;
 
-    let color: string | undefined = Colors.$textDefaultLight;
```
Comment (Inbal-Tish) [inline src/components/button/index.tsx:110]: Why did you remove Colors.$textDefaultLight as default?

File: src/components/button/index.tsx [hunk 3]
```diff
@@ -103,21 +103,32 @@ class Button extends PureComponent<Props, ButtonState> {
...
     if (isLink) {
-      color = linkColor || Colors.$textPrimary;
+      color = linkColor || contentColor || Colors.$textPrimary;
     } else if (outline) {
-      color = outlineColor || Colors.$textPrimary;
+      color = outlineColor || contentColor || Colors.$textPrimary;
     } else if (this.isIconButton) {
-      color = backgroundColor === 'transparent' ? undefined : Colors.$iconDefaultLight;
+      color = backgroundColor === 'transparent' ? undefined : contentColor || Colors.$textPrimary;
```
Comment (Inbal-Tish) [inline src/components/button/index.tsx:127]: Is this correct? Changing the default from iconDefaultLight to textPrimary?

File: src/components/button/index.tsx [hunk 4]
```diff
@@ -103,21 +103,32 @@ class Button extends PureComponent<Props, ButtonState> {
...
     }
 
-    if (disabled && (isLink || outline)) {
-      return Colors.$textDisabled;
+    if (disabled) {
+      return isLink || outline || fullWidth ? Colors.$textDisabled : Colors.white;
```
Comment (Inbal-Tish) [inline src/components/button/index.tsx:131]: Is this correct? The color of the disabled button is white?

Comment (Inbal-Tish) [reply src/components/button/index.tsx:131]: I'm wondering about disabled color in design kits and dark mode. Is it always white?

**Comment by adids1221** [comment]
Closing this PR, it's seems that we need the logic only in the private repo.
